--- ./src/linuxthreads.c	2008-03-28 15:38:57.000000000 -0700
+++ ../../coredumper-1.2.1/./src/linuxthreads.c	2019-05-13 11:04:18.333905537 -0700
@@ -39,7 +39,7 @@
 #endif
 
 #include <asm/stat.h>
-#include <sched.h>
+#include <linux/sched.h>
 #include <signal.h>
 #include <stdlib.h>
 #include <string.h>
@@ -49,7 +49,7 @@
 #include <asm/fcntl.h>
 #include <asm/posix_types.h>
 #include <asm/types.h>
-#include <linux/dirent.h>
+#include <dirent.h>
 
 #include "linux_syscall_support.h"
 #include "thread_lister.h"
