--- ./configure.ac.orig	2017-12-09 09:43:20.000000000 -0800
+++ ./configure.ac	2018-01-29 17:31:48.000000000 -0800
@@ -370,8 +370,8 @@
                [perftools_cv_sized_deallocation_result],
                [AC_LANG_PUSH(C++)
                 OLD_CXXFLAGS="$CXXFLAGS"
-                CXXFLAGS="$CXXFLAGS -fsized-deallocation"
-                AC_COMPILE_IFELSE([AC_LANG_PROGRAM(
+                CXXFLAGS="$CXXFLAGS -fsized-deallocation -O0 -fno-builtin"
+                AC_LINK_IFELSE([AC_LANG_PROGRAM(
                     [[#include <new>]],
                     [[(::operator delete)(0, 256)]])],
                  perftools_cv_sized_deallocation_result=yes,
@@ -385,18 +385,21 @@
 AC_CACHE_CHECK([if C++ compiler supports std::align_val_t without options],
                [perftools_cv_have_align_val_t],
                [AC_LANG_PUSH(C++)
+                OLD_CXXFLAGS="$CXXFLAGS"
+                CXXFLAGS="$CXXFLAGS -O0 -fno-builtin"
                 AC_LINK_IFELSE([AC_LANG_PROGRAM(
                     [[#include <new>]],
                     [[(::operator delete)((::operator new)(256, std::align_val_t(16)), std::align_val_t(16))]])],
                  perftools_cv_have_align_val_t=yes,
                  perftools_cv_have_align_val_t=no)
+                CXXFLAGS="$OLD_CXXFLAGS"
                 AC_LANG_POP(C++)])
 
 AC_CACHE_CHECK([if C++ compiler supports -faligned-new],
                [perftools_cv_have_f_aligned_new],
                [AC_LANG_PUSH(C++)
                 OLD_CXXFLAGS="$CXXFLAGS"
-                CXXFLAGS="$CXXFLAGS -faligned-new"
+                CXXFLAGS="$CXXFLAGS -faligned-new -O0 -fno-builtin"
                 AC_LINK_IFELSE([AC_LANG_PROGRAM(
                     [[#include <new>]],
                     [[(::operator delete)((::operator new)(256, std::align_val_t(16)), std::align_val_t(16))]])],
