--- a/tools/clang/tools/include-what-you-use/iwyu_include_picker.cc 2017-08-07 16:41:03.193267222 -0700
+++ b/tools/clang/tools/include-what-you-use/iwyu_include_picker.cc	2017-08-07 16:41:12.603327180 -0700
@@ -123,6 +123,13 @@ const IncludeMapEntry libc_symbol_map[] = {
   { "ino_t", kPrivate, "<sys/stat.h>", kPublic },
   { "int8_t", kPrivate, "<sys/types.h>", kPublic },
   { "int8_t", kPrivate, "<stdint.h>", kPublic },
+  { "uint8_t", kPrivate, "<stdint.h>", kPublic },
+  { "int16_t", kPrivate, "<stdint.h>", kPublic },
+  { "uint16_t", kPrivate, "<stdint.h>", kPublic },
+  { "int32_t", kPrivate, "<stdint.h>", kPublic },
+  { "uint32_t", kPrivate, "<stdint.h>", kPublic },
+  { "int64_t", kPrivate, "<stdint.h>", kPublic },
+  { "uint64_t", kPrivate, "<stdint.h>", kPublic },
   { "intptr_t", kPrivate, "<stdint.h>", kPublic },
   { "intptr_t", kPrivate, "<unistd.h>", kPublic },
   { "key_t", kPrivate, "<sys/types.h>", kPublic },
@@ -288,6 +295,7 @@ const IncludeMapEntry libc_include_map[] = {
   { "<bits/param.h>", kPrivate, "<sys/param.h>", kPublic },
   { "<bits/poll.h>", kPrivate, "<sys/poll.h>", kPrivate },
   { "<bits/poll2.h>", kPrivate, "<sys/poll.h>", kPrivate },
+  { "<bits/local_lim.h>", kPrivate, "<limits.h>", kPublic },
   { "<bits/posix1_lim.h>", kPrivate, "<limits.h>", kPublic },
   { "<bits/posix2_lim.h>", kPrivate, "<limits.h>", kPublic },
   { "<bits/posix_opt.h>", kPrivate, "<unistd.h>", kPublic },
